overlord/snapstate/handlers.go

```diff
@@ -1365,6 +1485,22 @@ func (m *SnapManager) undoCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	// if any migration action was taken, try to undo it and rewrite state
+	if snapsup.MigratedToExposedHome {
+		if err := m.backend.RemoveExposedSnapHome(snapsup.InstanceName()); err != nil {
+			return err
+		}
+
+		snapsup.MigratedToExposedHome = false
+		snapst.MigratedToExposedHome = false
+
+		st.Lock()
+		err = writeMigrationStatus(t, snapst, snapsup)
+		st.Unlock()
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
do we need to anything here about the DisableHomeMigration is  true case?

Reply:
As it is right now, the copy data doesn't disable the home migration so we don't need to deal with it here. It only has the forward path and the undo for the hidden migration (which would be triggered if the user disables the experimental flag and refreshes)

Reply:
So in the "copy-data" task we do the forward path for both migrations and the revert for the hidden migration, so when undoing it we only check those flags (lines 1488, 1505 and 1512). Currently we only disable the HOME migration in "unlink-current-snap", so we only check the DisableHomeMigration flag in undoUnlinkCurrentSnap (line 1134) and write the seq file accordingly (using `writeMigrationStatus`). If you're talking about writing seq file in the "happy" path, then the migration flags (including that one) are checked in "linkSnap" and set there.